components/sidebar/sidebar_category_list/index.ts

```diff
@@ -8,9 +8,27 @@ import {getOrderedChannelIds} from 'mattermost-redux/selectors/entities/channels
 import {getSidebarPreferences} from 'mattermost-redux/selectors/entities/preferences';
 import {GlobalState} from 'mattermost-redux/types/store';
 import {GenericAction} from 'mattermost-redux/types/actions';
+import {memoizeResult} from 'mattermost-redux/utils/helpers';
 
 import SidebarCategoryList from './sidebar_category_list';
 
+type ChannelCategory = {
+    type: string;
+    name: string;
+    items: string[];
+}
+
+function getCategoryFromChannel(channelCategories: ChannelCategory[]) {
```

Comment:
This is almost correct, but you need to call `memoizeResult` out here to make `getChannelFromCategory` memoized. Currently, you create a new memoized function in `mapStateToProps` which gets thrown away as soon as `mapStateToProps` has finished, so no memoization occurs.

```typescript
const getCategoryFromChannel.= memoizeResult((channelCategories: ChannelCategory[]) {
    // ...
}
```